packages/mui-utils/src/getReactElementRef/getReactElementRef.ts

```diff
@@ -5,16 +5,14 @@ import * as React from 'react';
  * It will throw runtime error if the element is not a valid React element.
  *
  * @param element React.ReactElement
- * @returns React.Ref<any> | null | undefined
+ * @returns React.Ref<any> | null
  */
-export default function getReactElementRef(
-  element: React.ReactElement,
-): React.Ref<any> | null | undefined {
+export default function getReactElementRef(element: React.ReactElement): React.Ref<any> | null {
   // 'ref' is passed as prop in React 19, whereas 'ref' is directly attached to children in older versions
   if (parseInt(React.version, 10) >= 19) {
-    return element.props?.ref;
+    return (element?.props as any)?.ref || null;
```

Comment:
Apart from making it return `null` when there's no ref, two other changes in this line:
- `element.props` is unknown so we need to cast it. With React 19 types, TS reports this error: `Property 'ref' does not exist on type '{}'`
- Optional chaining added to `element` so it doesn't fail in test cases where the value passed in undefined (`getReactElementRef()`)